app/models/concerns/reassociatable.rb

```diff
@@ -45,11 +48,14 @@ def update_child_objects
 
       values_to_update = check_headers(child_headers, row)
       update_child_values(values_to_update, co, row, index)
+      index + 1 if index + 1 - start_index > BatchProcess::BATCH_LIMIT
```

Comment:
You need the return here too, but with [parents_needing_update, parent_destination_map, index+1]